test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlWriterTests.cs

```diff
@@ -2460,6 +2460,77 @@ private void WriteAndVerifyXml(IEdmModel model, string expected, CsdlTarget targ
             }
         }
 
+        [Fact]
+        public void VerifyKeyAliasForComplexPropertiesIsSupported()
+        {
+            // Arrange
+            var model = new EdmModel();
+            var entity = new EdmEntityType("NS1", "Product");
+            var complexType = new EdmComplexType("NS1", "ExampleComplexObject");
+            complexType.AddStructuralProperty("Name", EdmCoreModel.Instance.GetString(false));
+            var entityId = entity.AddStructuralProperty("ExampleObject", new EdmComplexTypeReference(complexType, isNullable: false));
+            entity.AddKeys(entityId);
+            entity.AddKeyAlias(new KeyValuePair<IEdmStructuralProperty, string>(entityId, "ExampleObjectName"));

```

Comment:
This `entityId` is a reference to the complex property `ExampleObject`. However, according to the spec, the alias must reference a property of a complex property. It must not be a reference to the complex property itself ([the spec](https://docs.oasis-open.org/odata/odata-csdl-json/v4.01/odata-csdl-json-v4.01.html#sec_Key) says that a key property must primitive).

Reply:
So for example, the alias may reference the `Name` of the `ExampleObject` property, but it may not be a reference to `ExampleObject` itself, which is what seems to be happening here.

Reply:
<code> ExampleObject </code> is the name of the property in the declaring Type. and the type of <code> ExampleObject </code> is the complex type. <code> ExampleComplexObject </code>.

Reply:
```xml
<EntityType Name="Category">
  <Key>
    <PropertyRef Name="Info/ID" Alias="EntityInfoID" />
  </Key>
  <Property Name="Info" Type="Sales.EntityInfo" Nullable="false" />
  <Property Name="Name" Type="Edm.String" />
</EntityType>

 

<ComplexType Name="EntityInfo">

  <Property Name="ID" Type="Edm.Int32" Nullable="false" />
  <Property Name="Created" Type="Edm.DateTimeOffset" />
</ComplexType>
```

Reply:
aah I see what you mean.